modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/ManagementToolbar/index.tsx

```diff
@@ -78,6 +78,13 @@ export function ManagementToolbar({
 		onClose: () => setVisibleModal(false),
 	});
 
+	const [disabled, setDisabled] = useState(!hasPublishPermission);
+
+	function onClick() {
+		onSubmit(false);
+		setDisabled(hasPublishPermission);
+	}
```

Comment:
Hey @vngoliveira, I understand your idea, I just wouldn't change this `hasPublishPermission` condition because we can introduce some bugs with that, we can disable the button for 1 second, which is the time for this onSubmit to execute (see line `175` in `EditObjectDetails.tsx` ), but if for some reason the sending fails, we must enable the button again, the best way I found for this was:

```tsx
	const [disabled, setDisabled] = useState(false);

	const onPublish = () => {
		setDisabled(true);

		onSubmit(false);

		setTimeout(() => {
			setDisabled(false);
		}, 1000);
	};
```
And in the disabled button prop, we keep the old logic plus this new one:

```tsx
disabled={!hasPublishPermission || disabled}
```

Reply:
Yes, it works and looks better than the other solution. Thank you, @danvb112!